components/idp-mgt/org.wso2.carbon.identity.metadata.saml/src/main/java/org/wso2/carbon/identity/metadata/saml/builder/DefaultIDPMetadataBuilder.java

```diff
@@ -0,0 +1,230 @@
+        entityDescriptor.setNoNamespaceSchemaLocation("");
+        return entityDescriptor;
+    }
+
+    public IDPSSODescriptor buildIDPSSODescriptor() throws MetadataException {
+        IDPSSODescriptor idpSsoDesc = BuilderUtil
+                .createSAMLObject(IDPMetadataConstant.IDP_METADATA_SAML2, ConfigElements.IDPSSO_DESCRIPTOR, "");
+        return idpSsoDesc;
+    }
+
+    public void buildValidityPeriod(IDPSSODescriptor idpSsoDesc) throws MetadataException {
+        char unit = 'h'; // Default unit is hours
+        idpSsoDesc.setValidUntil(validityPeriod(1, unit));
+    }
+
+    public void buildSupportedProtocol(IDPSSODescriptor idpSsoDesc) throws MetadataException {
+        idpSsoDesc.addSupportedProtocol(IDPMetadataConstant.SUPPORTED_PROTOCOL_SAML2);
+    }
+
+    public void buildKeyDescriptor(EntityDescriptor entityDescriptor) throws MetadataException {
+        CryptoProvider cryptoProvider = new MetadataCryptoProvider();
+        cryptoProvider.signMetadata(entityDescriptor);
+    }
+
+    public String marshallDescriptor(EntityDescriptor entityDescriptor) throws MetadataException {
+
+        DocumentBuilderFactory factory = IdentityUtil.getSecuredDocumentBuilderFactory();
+        DocumentBuilder builder;
+        try {
+            builder = factory.newDocumentBuilder();
+        } catch (ParserConfigurationException e) {
+            throw new MetadataException("Error while creating the document.", e);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Marshalling the metadata element contents");
+        }
+        Document document = builder.newDocument();
+        Marshaller out = org.opensaml.xml.Configuration.getMarshallerFactory().getMarshaller(entityDescriptor);
+
+        try {
+            out.marshall(entityDescriptor, document);
+        } catch (MarshallingException e) {
+            throw new MetadataException("Error while marshalling the descriptor.", e);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Marshalling metadata completed.");
+        }
+        org.apache.xml.security.Init.init();
+
+        Transformer transformer;
+        StreamResult streamResult;
+        StringWriter stringWriter = new StringWriter();
+        try {
+            transformer = TransformerFactory.newInstance().newTransformer();
+
+            streamResult = new StreamResult(stringWriter);
+            DOMSource source = new DOMSource(document);
+            transformer.transform(source, streamResult);
+            stringWriter.close();
+            return stringWriter.toString();
+        } catch (TransformerConfigurationException e) {
+            log.error("Error Occurred while creating XML transformer", e);
+        } catch (IOException e) {
+            log.error("Error Occurred while creating XML transformer", e);
+        } catch (TransformerException e) {
+            log.error("Error Occurred while creating XML transformer", e);
+        }
```

Comment:
Shouldn't we throw the exceptions after wrapping the original exception? Otherwise `return stringWriter.toString();` might return a malformed string.